ocaml/xapi/xapi_vif.ml

```diff
@@ -55,12 +56,18 @@ let refresh_filtering_rules ~__context ~self =
 
 (* This function moves a dom0 vif device from one bridge to another, without involving the guest,
  * so it also works on guests that do not support hot(un)plug of VIFs. *)
-let move ~__context ~network vif =
-	debug "Moving VIF %s to network %s" (Db.VIF.get_uuid ~__context ~self:vif)
+let move ~__context ~network ~self =
```

Comment:
You have reused and existing internal function and added assertions to it. These new assertions are therefore also  executed on the existing code paths, which worries me. I would prefer to not change the existing code paths at all. You can do that by renaming the existing `move` function to `move_internal` and adding a new `move` function that contains the asserts and a call to `move_internal`.